http/middleware.go

```diff
@@ -152,3 +235,72 @@ func JSONWebKey(key string) AuthorizationOption {
 		return nil
 	}
 }
+
+// JWKSClient allows to provide a custom jwks.Client that will be
+// used when fetching the JSON Web Key Set with which the JWT
+// will be verified.
+// The JSONWebKey option takes precedence. If a web key is already
+// provided through the JSONWebKey option, the JWKS client will
+// not be used at all.
+func JWKSClient(client *jwks.Client) AuthorizationOption {
+	return func(params *AuthorizationParams) error {
+		params.JWKSClient = client
+		return nil
+	}
+}
+
+// A cache to store JSON Web Keys.
+type jwkCache struct {
+	mu      sync.RWMutex
+	entries map[string]*cacheEntry
+}
+
+// Each entry in the JWK cache has a value and an expiration date.
+type cacheEntry struct {
+	value     *clerk.JSONWebKey
+	expiresAt time.Time
+}
+
+// Get fetches the JSON Web Key for the provided key, unless the
+// entry has expired.
+func (c *jwkCache) Get(key string) *clerk.JSONWebKey {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+	entry, ok := c.entries[key]
+	if !ok || entry == nil {
+		return nil
+	}
+	if entry.expiresAt.Before(time.Now().UTC()) {
+		return nil
+	}
+	return entry.value
+}
+
+// Set stores the JSON Web Key in the provided key and sets the
+// expiration date.
+func (c *jwkCache) Set(key string, value *clerk.JSONWebKey, expiresAt time.Time) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	c.entries[key] = &cacheEntry{
+		value:     value,
+		expiresAt: expiresAt,
+	}
+}
+
+var cacheInit sync.Once
+
+// A "singleton" JWK cache for the package.
+var cache *jwkCache
+
+// GetCache returns the library's default cache singleton.
+// Please note that the returned Cache is a package-level variable.
+// Using the package with more than one Clerk API secret keys might
+// require to use different Clients with their own Cache.
+func GetCache() *jwkCache {
```

Comment:
:question: Do we need these functions to be exported? Maybe for testing?

Reply:
We don't. Updated.